pkg/secrets/factory.go

```diff
@@ -168,47 +195,45 @@ func CreateSecretProvider(managerType ProviderType) (Provider, error) {
 }
 
 // GetSecretsPassword returns the password to use for encrypting and decrypting secrets.
-// It will attempt to retrieve it from the environment variable TOOLHIVE_SECRETS_PASSWORD.
-// If the environment variable is not set, it will prompt the user to enter a password.
+// It will attempt to retrieve from the OS keyring.
+// If not available, it will fail with an error.
 func GetSecretsPassword() ([]byte, error) {
-	// First, attempt to load the password from the environment variable.
-	password := []byte(os.Getenv(PasswordEnvVar))
-	if len(password) > 0 {
-		return password, nil
-	}
-
-	// If not present, attempt to load the password from the OS keyring.
+	// Attempt to load the password from the OS keyring.
 	keyringSecret, err := keyring.Get(keyringService, keyringService)
 	if err == nil {
 		return []byte(keyringSecret), nil
 	}
 
 	// We need to determine if the error is due to a lack of keyring on the
 	// system or if the keyring is available but nothing was stored.
-	keyringAvailable := errors.Is(err, keyring.ErrNotFound)
-
-	// We cannot ask for a password in a detached process.
-	// We should never trigger this, but this ensures that if there's a bug
-	// then it's easier to find.
-	if process.IsDetached() {
-		return nil, fmt.Errorf("detached process detected, cannot ask for password")
-	}
+	if errors.Is(err, keyring.ErrNotFound) {
+
+		// Keyring is available but no password stored - this should only happen during setup
+		// We cannot ask for a password in a detached process.
+		// We should never trigger this, but this ensures that if there's a bug
+		// then it's easier to find.
+		if process.IsDetached() {
+			return nil, fmt.Errorf("detached process detected, cannot ask for password")
+		}
 
-	// If the keyring is not available, prompt the user for a password.
-	password, err = readPasswordStdin()
-	if err != nil {
-		return nil, fmt.Errorf("failed to read password: %w", err)
-	}
+		// Prompt for password during setup
```

Comment:
I wonder if we need a more robust mechanism to ensure that this part of the code is not executed outside of the CLI. Let's chat tomorrow.